connector/profiler/src/main/scala/org/apache/spark/executor/profiler/ExecutorJVMProfiler.scala

```diff
@@ -42,14 +42,15 @@ private[spark] class ExecutorJVMProfiler(conf: SparkConf, executorId: String) ex
   private val profilerLocalDir = conf.get(EXECUTOR_PROFILING_LOCAL_DIR)
   private val writeInterval = conf.get(EXECUTOR_PROFILING_WRITE_INTERVAL)
 
-  private val appId = try {
-    conf.getAppId
-  } catch {
-    case _: NoSuchElementException => "local-" + System.currentTimeMillis
+  // app_id and app_attempt_id is unavailable during drvier plugin initialization
+  private def getAppId: Option[String] = conf.getOption("spark.app.id")
```

Comment:
@parthchandra I hit the app id unavailable issue when adding the driver plugin support, and fixed it by deferring DFS dir creating.